diff -ru a/dlib/cmake_utils/find_blas.cmake b/dlib/cmake_utils/find_blas.cmake
--- a/dlib/cmake_utils/find_blas.cmake
+++ b/dlib/cmake_utils/find_blas.cmake
@@ -66,8 +66,8 @@
    
    # First, search for libraries via pkg-config, which is the cleanest path
    find_package(PkgConfig)
-   pkg_check_modules(BLAS_REFERENCE cblas)
-   pkg_check_modules(LAPACK_REFERENCE lapack)
+   pkg_check_modules(BLAS_REFERENCE flexiblas)
+   pkg_check_modules(LAPACK_REFERENCE flexiblas)
    # Make sure the cblas found by pkgconfig actually has cblas symbols.
    SET(CMAKE_REQUIRED_LIBRARIES "${BLAS_REFERENCE_LDFLAGS}")   
    CHECK_FUNCTION_EXISTS(cblas_ddot PKGCFG_HAVE_CBLAS)
diff -ru a/dlib/cmake_utils/set_compiler_specific_options.cmake b/dlib/cmake_utils/set_compiler_specific_options.cmake
--- a/dlib/cmake_utils/set_compiler_specific_options.cmake
+++ b/dlib/cmake_utils/set_compiler_specific_options.cmake
@@ -2,26 +2,6 @@
 cmake_minimum_required(VERSION 3.10.0)
 
 
-# Check if we are being built as part of a pybind11 module. 
-if (COMMAND pybind11_add_module)
-   # For python users, enable SSE4 and AVX if they have these instructions.
-   include(${CMAKE_CURRENT_LIST_DIR}/check_if_sse4_instructions_executable_on_host.cmake)
-   if (SSE4_IS_AVAILABLE_ON_HOST)
-      set(USE_SSE4_INSTRUCTIONS ON CACHE BOOL "Compile your program with SSE4 instructions")
-   endif()
-   include(${CMAKE_CURRENT_LIST_DIR}/check_if_avx_instructions_executable_on_host.cmake)
-   if (AVX_IS_AVAILABLE_ON_HOST)
-      set(USE_AVX_INSTRUCTIONS ON CACHE BOOL "Compile your program with AVX instructions")
-   endif()
-   include(${CMAKE_CURRENT_LIST_DIR}/check_if_neon_available.cmake)
-   if (ARM_NEON_IS_AVAILABLE)
-      set(USE_NEON_INSTRUCTIONS ON CACHE BOOL "Compile your program with ARM-NEON instructions")
-   endif()
-endif()
-
-
-
-
 set(gcc_like_compilers GNU Clang  Intel)
 set(intel_archs x86_64 i386 i686 AMD64 amd64 x86)
 
diff -ru a/tools/python/CMakeLists.txt b/tools/python/CMakeLists.txt
--- a/tools/python/CMakeLists.txt
+++ b/tools/python/CMakeLists.txt
@@ -28,7 +28,7 @@
 
 
 
-add_subdirectory(../../dlib/external/pybind11 pybind11_build)
+find_package(pybind11 CONFIG REQUIRED)
 add_subdirectory(../../dlib dlib_build)
 
 add_definitions(-DDLIB_VERSION=${DLIB_VERSION})
@@ -67,7 +67,7 @@
    list(APPEND python_srcs src/gui.cpp)
 endif()
 
-pybind11_add_module(_dlib_pybind11 ${python_srcs})
+pybind11_add_module(_dlib_pybind11 NO_EXTRAS ${python_srcs})
 target_link_libraries(_dlib_pybind11 PRIVATE dlib::dlib)
 
 # When invoked from setup.py CMAKE_LIBRARY_OUTPUT_DIRECTORY will be set.  But when you are just building via say
